index.js

```diff
@@ -0,0 +1,292 @@
+    
+        let Itime = note_time
+        Itime = Itime.replace(":",'')
+        Itime = Number(Itime)
+
+        Date.prototype.toDateInputValue = (function() {
+            let local = new Date(this);
+            local.setMinutes(this.getMinutes() - this.getTimezoneOffset());
+            return local.toJSON().slice(0,10);
+        
+        
+        });
+
+        let dateNow = new Date().toDateInputValue();
+        dateNow = dateNow.replace("-" , '')
+        dateNow = dateNow.replace("-",'')
+        dateNow = Number(dateNow)
+
+        let Idate = note_date
+        Idate = Idate.replace("-",'')
+        Idate = Idate.replace("-",'')
+        Idate = Number(Idate)
+      
+         if(Idate === dateNow && Itime < timeNow){return false;}
+        
+        
+    }
+    function validateDate(note_date)  {
+        Date.prototype.toDateInputValue = (function() {
+            let local = new Date(this);
+            local.setMinutes(this.getMinutes() - this.getTimezoneOffset());
+            return local.toJSON().slice(0,10);
+        
+        
+        });
+
+        let dateNow = new Date().toDateInputValue();
+        dateNow = dateNow.replace("-" , '')
+        dateNow = dateNow.replace("-",'')
+        dateNow = Number(dateNow)
+
+        let Idate = note_date
+        Idate = Idate.replace("-",'')
+        Idate = Idate.replace("-",'')
+        Idate = Number(Idate)
+      
+         if(Idate < dateNow){ return false;}
+    }
+   // local storage functions 
+
+    function saveToLocalStorage(key, value) {
+        localStorage.setItem(key, JSON.stringify(value));
+      }
+
+      function init() {
+        arrayOfNotes = JSON.parse(localStorage.getItem("stickyNotes")) || [];
+        draw(arrayOfNotes);
+      }
+      init();
+
+function sortBy(value,arr) {
+    switch(value){
+        case ("All"):
+            draw(arr);
+            break;
+        case ("ToDo"):
+            sortToDo(arr)
+            break;
+        case ("Today"):
+            sortToday(arr) 
+            break;
+        case ("Completed"):
+            sortCompleted(arr)
+            break;
+            
+    }
+}
+
+function sortToday(arr) {
+    Date.prototype.toDateInputValue = (function() {
+        let local = new Date(this);
+     
+        local.setMinutes(this.getMinutes() - this.getTimezoneOffset());
+        return local.toJSON().slice(0,10);
+    });   
+    clearBoard()
+    for (let index = 0; index < arr.length; index++) {
+        let dateNow = new Date().toDateInputValue();
+let dateYear = dateNow.slice(0, 4);
+let dateMonth = dateNow.slice(5, 7);
+let dateDay = dateNow.slice(8, 10);
+dateNow = (dateDay + dateMonth + dateYear)
+        dateNow = Number(dateNow)
+    console.log(dateNow)
+       let currentNote = arr[index]
+        let Idate = currentNote.note_date
+    Idate = Idate.replace("-",'')
+    Idate = Idate.replace("-",'')
+    Idate = Number(Idate)
+    console.log(Idate)
+     if(Idate === dateNow ){
```

Comment:
1. You don't need to convert the string to a number to compare them
2. Another and more correct way to do this is tou use the Date objects and just clear the parts of the Date object that you don't care and then compare them

Reply:
i need it because the function date now is giving me a time in a number